From: Andi Kleen <ak@suse.de>
Date: Mon, 10 Oct 2005 23:03:39 +0000 (+0200)
Subject:     [PATCH] x86_64: Fix change_page_attr cache flushing
X-Git-Tag: v2.6.14-rc4
X-Git-Url: http://www.kernel.org/git/gitweb.cgi?p=linux/kernel/git/torvalds/linux-2.6.git;a=commitdiff;h=094804c5a132f04c12dd4902ee15c64362e5c1af

  [PATCH] x86_64: Fix change_page_attr cache flushing
  
  Noticed by Terence Ripperda
  
  Undo wrong change in global_flush_tlb. We need to flush the caches in all
  cases, not just when pages were reverted. This was a bogus optimization
  added earlier, but it was wrong.
  
  Signed-off-by: Andi Kleen <ak@suse.de>
  Signed-off-by: Linus Torvalds <torvalds@osdl.org>
---

--- a/arch/x86_64/mm/pageattr.c
+++ b/arch/x86_64/mm/pageattr.c
@@ -220,8 +220,6 @@ void global_flush_tlb(void)
 	down_read(&init_mm.mmap_sem);
 	df = xchg(&df_list, NULL);
 	up_read(&init_mm.mmap_sem);
-	if (!df)
-		return;
 	flush_map((df && !df->next) ? df->address : 0);
 	for (; df; df = next_df) { 
 		next_df = df->next;
